core/src/main/java/org/apache/iceberg/rest/responses/FetchPlanningResultResponse.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.responses;
+
+import java.util.List;
+import javax.annotation.Nullable;
+import org.apache.iceberg.DeleteFile;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.rest.PlanStatus;
+import org.apache.iceberg.rest.RESTResponse;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface FetchPlanningResultResponse extends RESTResponse {
+  PlanStatus planStatus();
+
+  @Nullable
+  List<String> planTasks();
+
+  @Nullable
+  List<FileScanTask> fileScanTasks();
+
+  @Nullable
+  List<DeleteFile> deleteFiles();
+
+  @Override
+  default void validate() {
+    Preconditions.checkArgument(
+        planStatus() != null, "invalid, status can not be null: ", planStatus());
```

Comment:
Error messages should use sentence case. We also prefer direct statements rather than sentences, like `"Invalid status: null"`.

Reply:
this validation isn't really needed because `planStatus()` is required in the API definition so you won't even be able to construct an instance of this class through the builder. I think this will get more clear if you add the request/response parsers.

Reply:
Makes sense thanks